lib/Storage.php

```diff
@@ -164,7 +164,9 @@ public function getDocumentByShareToken($token, $fileId = null) {
 			if ($node->getType() == FileInfo::TYPE_FILE) {
 				/** @var \OCP\Files\Node|null $document */
 				$document = $node;
-			} elseif ($node->getType() == FileInfo::TYPE_FOLDER && $fileId !== null) {
+			} elseif ($fileId !== null) {
```

Comment:
At this point we know that the resource type is not TYPE_FILE. `getType` can only return 2 values, so we know that `$node` must be of type `TYPE_FOLDER`. Therefore that check is removed.